test/test_poisson_solvers.jl

```diff
@@ -136,9 +135,19 @@ end
 ##### Vertically stretched Poisson solver
 #####
 
+get_grid_size(TX, TY, TZ, Nx, Ny, Nz) = (Nx, Ny, Nz)
+get_grid_size(::Type{Flat}, TY, TZ, Nx, Ny, Nz) = (Ny, Nz)
+get_grid_size(TX, ::Type{Flat}, TZ, Nx, Ny, Nz) = (Nx, Nz)
```

Comment:
I left a comment but it might have disappeared.  

Should we have a `get_grid_size` for the case of `Flat` in both `x` and `y`?

Reply:
Well, we don't test that case in `test_poisson_solvers.jl`. We could write that in. But when a model is 1D there's no pressure field (and no vertical momentum equation, since `w_z = 0` and thus `w = constant`). We should support this case for modeling, but I don't think we should do a poisson solve when it's 1D. Instead we shouldn't have a poisson solver at all. I think properly supporting 1D (making it as cheap as possible) requires a bit more work.

Reply:
Thanks.  I am looking forward to using a stretched grid in a 1D plankton model, so I am all in favor of us supporting this. But you are right, since we don't test it there is no need to include it here.

Reply:
You can run a 1D plankton model by using `Periodic, Periodic, Bounded` topology with 1 grid point in x and y. Or do you need the model to be high performance? There's no advection so you can use `advection=nothing`. You'll have horizontal halos that you don't need but because the model is 1D it's cheap. Unless you want to use this model for machine learning or data assimilation, I think you'll be able to do the science you want?